src/backend/gl/src/device.rs

```diff
@@ -1627,8 +1627,31 @@ impl d::Device<B> for Device {
         I: IntoIterator,
         I::Item: Borrow<pso::DescriptorSetCopy<'a, B>>,
     {
-        for _copy in copies {
-            unimplemented!()
+        for copy in copies {
+            let copy = copy.borrow();
+
+            let src_set = &copy.src_set;
+            let dst_set = &copy.dst_set;
+            if std::ptr::eq(src_set, dst_set) {
+                panic!("copying within same descriptor set is not currently supported");
+            }
+
+            let src_bindings = src_set.bindings.lock();
+            let mut dst_bindings = dst_set.bindings.lock();
+
+            let count = copy.count;
+
+            let src_start = copy.src_binding as usize + copy.src_array_offset;
```

Comment:
this seems wrong. If all the bindings are laid out with all the array elements flattened, then we'd need to sum up all the array counts for descriptor bindings `0 .. copy.src_binding`

Reply:
From what I can understand, these are the types of bindings OpenGL supports:
```rust
pub(crate) enum DescSetBindings {
    Buffer { /* ... */ },
    Texture(/* ... */),
    Sampler(/* ... */),
    SamplerDesc(/* ... */),
}
```
It seems OpenGL doesn't support array bindings?

Reply:
In that case, `src_array_offset` has to be zero, and we don't need to check it. We can have a "TODO: support array bindings" line

Reply:
Ok, I've updated the PR to add a `// TODO` for when GL gets support for array bindings.